finetune/adapter.py

```diff
@@ -135,10 +135,10 @@ def train(
 
     with torch.device("meta"):
         meta_model = GPT(model.config)
-        # estimated is too much of an optimistic estimate, left just for reference
-        estimated_flops = estimate_flops(meta_model) * micro_batch_size
-        fabric.print(f"Estimated TFLOPs: {estimated_flops * fabric.world_size / 1e12:.2f}")
-        x = torch.randint(0, 1, (micro_batch_size, model.config.block_size))
+        # estimated flops doesn't account for frozen weights, so it's not reported
+        mark_only_adapter_as_trainable(meta_model)
+        # TODO: this assumes that samples have a fixed length which is most likely false during finetuning
```

Comment:
I might try addressing this TODO in a follow-up. For now, I just documented this limitation